rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,127 @@
+---
+feature: meson_builds_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for the reference implementation of Nix.
```

Comment:
```suggestion
Use meson as an alternative build system for Nix.
```

This incorrectly suggests a distinction between Nix and its implementation.

Reply:
the idea is to allow nix (self compile) & meson to both compile the nix code. This would remove the requirement for a working nix installation to compile nix, which would greatly complicate building/use/development with the need to cross compile for other operating systems / architectures.

Reply:
> nix (self compile)

Can you tell me one benefit of providing a Nix-native build system for Nix? I don't see any practical advantage, just downsides.

(Interesting could be, perhaps a Nix library that converts ninja files into fine grained derivations, but that would be very much orthogonal to the RFC).

Reply:
Or do you mean that Nix is built in a Nix derivation? That case is not different from building with meson in a non-Nix-environment in a meaningful way, so this doesn't really need to be considered as a separate case.

Reply:
@p01arst0rm I don't think Eelcos point has to do with trying to build Nix with and without Nix. The Nix build of Nix uses autoconf today, it would use Meson with this RFC, I don't see any issues.

> This incorrectly suggests a distinction between Nix and its implementation.

I hope this will stop being true, but today it is true so I am fine with this revision.

Reply:
> This incorrectly suggests a distinction between Nix and its implementation.

I thought that it would be useful to distinguish them, because the Nix language can possibly be implemented in other ways. 

On the other hand, it should be obvious that there are no other implementations that we should care for the purposes of this RFC.